### PR TITLE
Change UX to set an asset profile as a benchmark 

### PR DESCRIPTION
**Pull request by M27afk**
- Added Checkbox as feature to set or unset a symbol as Benchmark.
- Modified controllers and funcitons to facilitate adding and removing Symbol from Benchmarks list.

@dtslvr 
[Issue](https://github.com/ghostfolio/ghostfolio/issues/2381)

### CODE REVIEW

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.component.ts [hunk 1]
```diff
@@ -152,6 +152,17 @@ export class AssetProfileDialog implements OnDestroy, OnInit {
       });
   }
 
+  public onUnsetBenchmark({ dataSource, symbol }: UniqueAsset) {
```
Comment (dtslvr) [inline apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.component.ts:155]: Can you move below `public onSubmit() {}` to keep the alphabetical order?

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.component.ts [hunk 2]
```diff
@@ -152,6 +152,17 @@ export class AssetProfileDialog implements OnDestroy, OnInit {
...
+    this.dataService
+      .deleteBenchmark({ dataSource, symbol })
+      .pipe(takeUntil(this.unsubscribeSubject))
+      .subscribe(() => {
```
Comment (dtslvr) [inline apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.component.ts:159]: I think the reload is disturbing and not needed here. What about:

```
.subscribe();
```

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.module.ts [hunk 3]
```diff
@@ -22,6 +23,7 @@ import { AssetProfileDialog } from './asset-profile-dialog.component';
     GfValueModule,
     MatButtonModule,
     MatDialogModule,
+    MatCheckboxModule,
```
Comment (dtslvr) [inline apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.module.ts:26]: Can you move up to keep the alphabetical order?

File: apps/client/src/app/services/data.service.ts [hunk 4]
```diff
@@ -467,6 +467,10 @@ export class DataService {
     return this.http.post(`/api/v1/benchmark`, benchmark);
   }
 
+  public deleteBenchmark(benchmark: UniqueAsset) {
```
Comment (dtslvr) [inline apps/client/src/app/services/data.service.ts:470]: Can you move up to keep the alphabetical order?

File: apps/api/src/app/benchmark/benchmark.controller.ts [hunk 5]
```diff
@@ -94,4 +94,41 @@ export class BenchmarkController {
       );
     }
   }
+
+  @Post('remove')
```
Comment (dtslvr) [inline apps/api/src/app/benchmark/benchmark.controller.ts:98]: Can you change this to `@Delete(':dataSource/:symbol')`?

File: apps/api/src/app/benchmark/benchmark.controller.ts [hunk 6]
```diff
@@ -94,4 +94,41 @@ export class BenchmarkController {
...
+  @UseGuards(AuthGuard('jwt'))
+  public async deleteBenchmark(@Body() { dataSource, symbol }: UniqueAsset) {
```
Comment (dtslvr) [inline apps/api/src/app/benchmark/benchmark.controller.ts:100]: `@Param('dataSource') dataSource: DataSource, @Param('symbol') symbol: string`

File: apps/api/src/app/benchmark/benchmark.service.ts [hunk 7]
```diff
@@ -245,6 +245,45 @@ export class BenchmarkService {
...
+  }: UniqueAsset): Promise<Partial<SymbolProfile>> {
+    const assetProfile = await this.prismaService.symbolProfile.findFirst({
+      where: {
+        dataSource,
+        symbol
+      }
+    });
+
+    if (!assetProfile) {
+      return;
```
Comment (dtslvr) [inline apps/api/src/app/benchmark/benchmark.service.ts:260]: I would return `null` here

File: apps/api/src/app/benchmark/benchmark.service.ts [hunk 8]
```diff
@@ -245,6 +245,45 @@ export class BenchmarkService {
...
+    }
+
+    let benchmarks =
+      ((await this.propertyService.getByKey(
+        PROPERTY_BENCHMARKS
+      )) as BenchmarkProperty[]) ?? [];
+
+    benchmarks = benchmarks.filter(
+      (item) => item.symbolProfileId !== assetProfile.id
```
Comment (dtslvr) [inline apps/api/src/app/benchmark/benchmark.service.ts:269]: Can you change to keep the style consistent?

```
({symbolProfileId}) => { return symbolProfileId !== assetProfile.id; }
```

**Review by dtslvr — CHANGES_REQUESTED** [review]
This already works as desired, @M27afk. I have added a few more comments.

**Review by M27afk — COMMENTED** [review]
@dtslvr 
- Replaced Post request with Delete.
- Arranged functions in alphabetical order.
- Added within filter function ```
  ({ symbolProfileId }) => {
        return symbolProfileId !== assetProfile.id;
      }```
- Removed reload function, instead added `isBenchmark` to change to `false`.

File: apps/api/src/app/benchmark/benchmark.controller.ts (hunk 6 above)
Comment (M27afk) [reply apps/api/src/app/benchmark/benchmark.controller.ts]: Resolved in latest commit.

File: apps/api/src/app/benchmark/benchmark.service.ts (hunk 7 above)
Comment (M27afk) [reply apps/api/src/app/benchmark/benchmark.service.ts]: Resolved in latest commit.

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.module.ts (hunk 3 above)
Comment (M27afk) [reply apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.module.ts:25]: Resolved in latest commit.

File: apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.component.ts (hunk 2 above)
Comment (M27afk) [reply apps/client/src/app/components/admin-market-data/asset-profile-dialog/asset-profile-dialog.component.ts]: Resolved in latest commit.

Instead added
` this.isBenchmark = false;`

File: apps/api/src/app/benchmark/benchmark.controller.ts (hunk 5 above)
Comment (M27afk) [reply apps/api/src/app/benchmark/benchmark.controller.ts]: Resolved in latest commit.

File: apps/api/src/app/benchmark/benchmark.service.ts (hunk 8 above)
Comment (M27afk) [reply apps/api/src/app/benchmark/benchmark.service.ts]: Resolved in latest commit.

**Review by dtslvr — APPROVED** [review]
Thank you for the changes, @M27afk! I have made some more minor improvements...

**Pull request merged**